diff --git a/src/wrappers/python/pygroupsig/libgroupsig_build.py b/src/wrappers/python/pygroupsig/libgroupsig_build.py
index c46f859..7d0a8ef 100644
--- a/src/wrappers/python/pygroupsig/libgroupsig_build.py
+++ b/src/wrappers/python/pygroupsig/libgroupsig_build.py
@@ -105,38 +105,40 @@ ffibuilder.cdef(groupsigcdef)
 
 c_header_file = path.Path("../../../src/include/groupsig.h").abspath()
 c_include_path = path.Path("../../../src/include").abspath()
-c_lib_path =  path.Path("../../../build/lib/libgroupsig-static.a").abspath()
-c_gl19_path =  path.Path("../../../build/lib/libgl19.a").abspath()
-c_bbs04_path =  path.Path("../../../build/lib/libbbs04.a").abspath()
-c_ps16_path =  path.Path("../../../build/lib/libps16.a").abspath()
-c_klap20_path =  path.Path("../../../build/lib/libklap20.a").abspath()
-c_kty04_path =  path.Path("../../../build/lib/libkty04.a").abspath()
-c_dl21_path =  path.Path("../../../build/lib/libdl21.a").abspath()
-c_dl21seq_path =  path.Path("../../../build/lib/libdl21seq.a").abspath()
-c_logger_path =  path.Path("../../../build/lib/liblogger.a").abspath()
-c_msg_path =  path.Path("../../../build/lib/libmsg.a").abspath()
-c_base64_path =  path.Path("../../../build/lib/libbase64.a").abspath()
-c_big_path =  path.Path("../../../build/lib/libbig.a").abspath()
-c_hash_path =  path.Path("../../../build/lib/libhash.a").abspath()
-c_pbcext_path =  path.Path("../../../build/lib/libpbcext.a").abspath()
-c_crypto_path =  path.Path("../../../build/lib/libgcrypto.a").abspath()
-c_math_path =  path.Path("../../../build/lib/libmath.a").abspath()
-c_sys_path =  path.Path("../../../build/lib/libsys.a").abspath()
-c_misc_path =  path.Path("../../../build/lib/libmisc.a").abspath()
-c_mcl_path =  path.Path("../../../build/external/lib/libmcl.a").abspath()
-c_mcl384_256_path = path.Path("../../../build/external/lib/libmclbn384_256.so").abspath()
-c_include_mcl_path = path.Path("../../../build/external/include/mcl").abspath()
-c_extlibs_path = path.Path("../../../build/external/lib").abspath()
+c_lib_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libgroupsig-static.a").abspath()
+c_gl19_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libgl19.a").abspath()
+c_bbs04_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libbbs04.a").abspath()
+c_ps16_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libps16.a").abspath()
+c_klap20_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libklap20.a").abspath()
+c_kty04_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libkty04.a").abspath()
+c_dl21_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libdl21.a").abspath()
+c_dl21seq_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libdl21seq.a").abspath()
+c_logger_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/liblogger.a").abspath()
+c_msg_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libmsg.a").abspath()
+c_base64_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libbase64.a").abspath()
+c_big_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libbig.a").abspath()
+c_hash_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libhash.a").abspath()
+c_pbcext_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libpbcext.a").abspath()
+c_crypto_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libgcrypto.a").abspath()
+c_math_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libmath.a").abspath()
+c_sys_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libsys.a").abspath()
+c_misc_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/lib/libmisc.a").abspath()
+c_mcl_path =  path.Path("/spirs_tee_sdk/build/libgroupsig/external/lib/libmcl.a").abspath()
+c_mcl384_256_path = path.Path("/spirs_tee_sdk/build/libgroupsig/external/lib/libmclbn384_256.so").abspath()
+c_include_mcl_path = path.Path("/spirs_tee_sdk/build/libgroupsig/external/include/mcl").abspath()
+c_extlibs_path = path.Path("/spirs_tee_sdk/build/libgroupsig/external/lib").abspath()
 
 # Specify sources and library dependencies
 ffibuilder.set_source("_groupsig",
                       r"""
                       #include "groupsig.h"
-                      #include "kty04.h"
-                      #include "klap20.h"
                       #include "gl19.h"
-                      #include "ps16.h"
                       #include "bbs04.h"
+                      #include "ps16.h"
+                      #include "klap20.h"
+                      #include "kty04.h"
+                      #include "dl21.h"
+                      #include "dl21seq.h"
                       """,
                       libraries=["stdc++","ssl","crypto"],
                       runtime_library_dirs=[
